packages/ts/file-router/src/vite-plugin/collectRoutesFromFS.ts

```diff
@@ -106,16 +111,11 @@ export default async function collectRoutesFromFS(
 
   [children, layout] = await Promise.all([
     Promise.all(
-      children.map(async (child) => {
-        let { file: f, layout: l } = child;
-        [f, l] = await Promise.all([checkFile(f, logger), checkFile(l, logger)]);
-
-        return {
-          ...child,
-          file: f,
-          layout: l,
-        };
-      }),
+      children.map(async (child) => ({
```

Comment:
What if instead of removing the filecheck here, we filter the array according to the results of `checkFile`?

Reply:
My reason for removing the `checkFile` call for the `file` was due to the fact that it's done earlier before adding it to the `children` entries. In case we want to add filtering to the pipeline before the map, we still don't need to call the `checkFile` on `file` as it is done earlier, unless we don't check the `file` earlier.